src/mc-fle2-payload-iev-v2.c

```diff
@@ -491,36 +404,49 @@ bool mc_FLE2IndexedRangeEncryptedValueV2_parse(mc_FLE2IndexedEncryptedValueV2_t
 
     CHECK_AND_RETURN(mc_reader_read_u8(&reader, &iev->fle_blob_subtype, status));
 
-    if (iev->fle_blob_subtype != MC_SUBTYPE_FLE2IndexedRangeEncryptedValueV2) {
-        CLIENT_ERR("mc_FLE2IndexedRangeEncryptedValueV2_parse expected "
-                   "fle_blob_subtype %d got: %" PRIu8,
-                   MC_SUBTYPE_FLE2IndexedRangeEncryptedValueV2,
+    if (iev->fle_blob_subtype != MC_SUBTYPE_FLE2IndexedEqualityEncryptedValueV2
+        && iev->fle_blob_subtype != MC_SUBTYPE_FLE2IndexedRangeEncryptedValueV2) {
+        CLIENT_ERR("mc_FLE2IndexedEncryptedValueV2_parse expected "
+                   "fle_blob_subtype MC_SUBTYPE_FLE2Indexed(Equality|Range)EncryptedValueV2 got: %" PRIu8,
                    iev->fle_blob_subtype);
         return false;
     }
 
+    if (iev->fle_blob_subtype == MC_SUBTYPE_FLE2IndexedEqualityEncryptedValueV2) {
+        iev->type = kTypeEquality;
+    } else {
+        iev->type = kTypeRange;
+    }
+
```

Comment:
I feel like we can combine the type setting the error checking to make something a little more readable, e.g.:
```suggestion
    if (iev->fle_blob_subtype == MC_SUBTYPE_FLE2IndexedEqualityEncryptedValueV2) {
        iev->type = kTypeEquality;
    } else if (iev->fle_blob_subtype == MC_SUBTYPE_FLE2IndexedRangeEncryptedValueV2) {
        iev->type = kTypeRange;
    } else {
        CLIENT_ERR("mc_FLE2IndexedEncryptedValueV2_parse expected "
                   "fle_blob_subtype MC_SUBTYPE_FLE2Indexed(Equality|Range)EncryptedValueV2 got: %" PRIu8,
                   iev->fle_blob_subtype);
        return false;    
    }

```